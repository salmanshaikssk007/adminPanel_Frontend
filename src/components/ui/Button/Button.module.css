.button {
  border: none;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  transition-duration: 0.4s;
  box-shadow: 1px 1px 8px 1px rgba(0, 0, 0, 0.24);
}
.button:active {
  transform: scale(0.8);
}
