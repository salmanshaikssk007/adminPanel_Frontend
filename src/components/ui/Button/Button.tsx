import React from 'react';
import Styles from './Button.module.css';

interface props {
  children: React.ReactNode;
  inputStyles?: any;
  clickEvent?: React.MouseEventHandler<HTMLButtonElement> | undefined;
  type?: 'button' | 'reset' | 'submit';
  setDisabled?: boolean;
}
const Button = (props: props) => {
  const combinedStyles = `${Styles.button} ${props.inputStyles}`;
  return (
    <button
      disabled={props.setDisabled}
      type={props.type}
      onClick={props.clickEvent}
      data-testid="button_ui"
      className={combinedStyles}>
      {props.children}
    </button>
  );
};

export default Button;
