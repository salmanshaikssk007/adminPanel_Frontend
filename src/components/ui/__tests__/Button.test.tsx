import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from '../Button/Button';

configure({ adapter: new Adapter() });

describe('<Button />', () => {
  test('test props as children', () => {
    const wrapper = shallow(<Button>children</Button>);
    expect(wrapper.text()).toBe('children');

    const button = wrapper.find('button');
    expect(button.type()).toBe('button');

    expect(wrapper).toMatchSnapshot();
  });
});
