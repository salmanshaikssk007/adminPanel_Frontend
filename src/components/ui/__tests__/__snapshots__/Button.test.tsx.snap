// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> test props as children 1`] = `
<button
  className="button undefined"
  data-testid="button_ui"
>
  children
</button>
`;
